index.js

```diff
@@ -25,14 +27,30 @@ module.exports = {
       name: 'ember-cli-babel',
       ext: 'js',
       toTree: function(tree) {
-        return require('broccoli-babel-transpiler')(tree, getBabelOptions(addon));
+        return require('broccoli-babel-transpiler')(tree, addon._getBabelOptions());
       }
     });
   },
 
   shouldIncludePolyfill: function() {
-    var options = getAddonOptions(this);
-    return options.includePolyfill === true;
+    var addonOptions = this._getAddonOptions();
+    var babelOptions = addonOptions.babel;
+    var customOptions = addonOptions['ember-cli-babel'];
+
+    if (customOptions && 'includePolyfill' in customOptions) {
+      return customOptions.includePolyfill === true;
```

Comment:
Good point RE, newer CLI's only doing the deprecation. Maybe you can leave an inline comment (so future selves don't see it's use here as an excuse to use it for other ember-CLI deprecations)? I don't feel strongly either way though...